### PR TITLE
(8133 ): Updated transit_vpc resource to add support for transit gateway in azure

### PR DESCRIPTION
**Pull request by zhiqiangjin**

### CODE REVIEW

File: aviatrix/resource_transit_vpc.go [hunk 1]
```diff
@@ -110,14 +115,31 @@ func resourceAviatrixTransitVpcCreate(d *schema.ResourceData, meta interface{})
...
-		tagList := d.Get("tag_list").([]interface{})
-		tagListStr := goaviatrix.ExpandStringList(tagList)
-		gateway.TagList = strings.Join(tagListStr, ",")
+
+	if cloudType == 1 {
+		if _, ok := d.GetOk("tag_list"); ok {
+			tagList := d.Get("tag_list").([]interface{})
+			tagListStr := goaviatrix.ExpandStringList(tagList)
+			gateway.TagList = strings.Join(tagListStr, ",")
+		}
```
Comment (arvindsree) [inline aviatrix/resource_transit_vpc.go]: if not aws show error that tags is not supported

File: aviatrix/resource_transit_vpc.go [hunk 2]
```diff
@@ -128,12 +150,17 @@ func resourceAviatrixTransitVpcCreate(d *schema.ResourceData, meta interface{})
 	}
 	haSubnet := d.Get("ha_subnet").(string)
 	haGwSize := d.Get("ha_gw_size").(string)
-	enableHybridConnection := d.Get("enable_hybrid_connection").(bool)
+	var enableHybridConnection bool
```
Comment (arvindsree) [inline aviatrix/resource_transit_vpc.go]: uninitialized can cause unexpected behaviour. Initalize to False

File: aviatrix/resource_transit_vpc.go [hunk 3]
```diff
@@ -403,26 +448,32 @@ func resourceAviatrixTransitVpcUpdate(d *schema.ResourceData, meta interface{})
...
+			} else {
+				err := client.DetachTransitGWForHybrid(transitGateway)
+				if err != nil {
+					return fmt.Errorf("failed to disable transit GW for Hybrid: %s", err)
+				}
 			}
 		}
+	} else {
+		if d.HasChange("enable_hybrid_connection") {
+			return fmt.Errorf("enable_hybrid_connection is not supported")
```
Comment (arvindsree) [inline aviatrix/resource_transit_vpc.go]: say enable_hybrid connection is only supported  for AWS cloud type 1

Comment (zhiqiangjin) [reply aviatrix/resource_transit_vpc.go]: agree

File: aviatrix/resource_transit_vpc.go (hunk 2 above)
Comment (zhiqiangjin) [reply aviatrix/resource_transit_vpc.go]: agree

**Comment by arvindsree** [comment]
We need to add acceptance tests for this

File: aviatrix/resource_transit_vpc.go (hunk 1 above)
Comment (zhiqiangjin) [reply aviatrix/resource_transit_vpc.go]: agree

**Pull request merged**